Feature/s3 c 1139 sproxyd batchdelete

rahulreddy opened the pull request:
## Description

Leverage Sproxyd's batch delete functionality to reduce orphans
### Motivation and context

Silent deletions could potentially create a lot of orphans in the data backend when the backend is unstable. This PR leverages Sproxyd's batch delete feature when cleaning up data when an object is being overwritten.

rahulreddy committed:
bugfix: S3C-1139 use batch delete in api methods

This commits makes the batch delete invocation to use callbacks
so that orphans are not created when sproxyd is unstable.

bert-e commented:
# Hello rahulreddy,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.5/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.4/feature/S3C-1139-sproxyd-batchdelete`
and `development/7.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.5/feature/S3C-1139-sproxyd-batchdelete origin/development/7.5
 $ git merge origin/w/7.4/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.5/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# History mismatch



Merge commit #730d8c666fed572ae71d6a4f04854e88b7149155 on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #1bee231bab3ef65e33bf7fe60b022a29467071a3 on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

rahulreddy commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.5/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.4/feature/S3C-1139-sproxyd-batchdelete`
and `development/7.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.5/feature/S3C-1139-sproxyd-batchdelete origin/development/7.5
 $ git merge origin/w/7.4/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.5/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

dora-korpar commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.5/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.4/feature/S3C-1139-sproxyd-batchdelete`
and `development/7.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.5/feature/S3C-1139-sproxyd-batchdelete origin/development/7.5
 $ git merge origin/w/7.4/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.5/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/S3C-1139-sproxyd-batchdelete origin/development/8.1
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/S3C-1139-sproxyd-batchdelete` into
`stabilization/7.4.4`
* [w/7.4/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.4...w/7.4/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.4` (pull request #1872)
* [w/7.5/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.5...w/7.5/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.5` (pull request #1873)
* [w/8.0/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.0...w/8.0/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.0` (pull request #1874)
* [w/8.1/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.1` (pull request #1875)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @rachedbenmustapha

* @rahulreddy

* @dora-korpar

dora-korpar commented:
@bert-e approve

bert-e commented:
# Not Author



I'm afraid I cannot do that, @dora-korpar:

Only the author of the pull request @rahulreddy can use this command.

Please **delete** the corresponding comment so I can move on.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @rachedbenmustapha

* @rahulreddy

* @dora-korpar

dora-korpar commented:
@bert-e bypass_author_approval

bert-e commented:
# Not authorized



I'm afraid I cannot do that, @dora-korpar:

> @bert-e bypass_author_approval


You don't have enough credentials to use `bypass_author_approval`.


Please **edit** or **delete** the corresponding comment so I can move on.

jonathan-gramain commented:
@bert-e bypass_author_approval

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/b06ce86e8dde165d1ede920886e00a1067509f6d) did not succeed in branch w/7.5/feature/S3C-1139-sproxyd-batchdelete.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/8f35a35625c1bd4145a75c146feb8a3e93f1ed99) did not succeed in branch w/8.1/feature/S3C-1139-sproxyd-batchdelete.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/69b1e2c8ae4f39018e0da2ba32c9ea461df8eaba) did not succeed in branch w/8.0/feature/S3C-1139-sproxyd-batchdelete.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

Please resolve the conflict on **the integration branch** (`w/8.1/feature/S3C-1139-sproxyd-batchdelete`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.1/feature/S3C-1139-sproxyd-batchdelete
 $ git pull  # or "git reset --hard origin/w/8.1/feature/S3C-1139-sproxyd-batchdelete"
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

dora-korpar commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

Please resolve the conflict on **the integration branch** (`w/8.1/feature/S3C-1139-sproxyd-batchdelete`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/8.1/feature/S3C-1139-sproxyd-batchdelete
 $ git pull  # or "git reset --hard origin/w/8.1/feature/S3C-1139-sproxyd-batchdelete"
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

dora-korpar commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.5/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.4/feature/S3C-1139-sproxyd-batchdelete`
and `development/7.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.5/feature/S3C-1139-sproxyd-batchdelete origin/development/7.5
 $ git merge origin/w/7.4/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.5/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/S3C-1139-sproxyd-batchdelete origin/development/8.1
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/S3C-1139-sproxyd-batchdelete` into
`stabilization/7.4.4`
* [w/7.4/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.4...w/7.4/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.4` (pull request #1880)
* [w/7.5/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.5...w/7.5/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.5` (pull request #1881)
* [w/8.0/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.0...w/8.0/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.0` (pull request #1882)
* [w/8.1/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.1` (pull request #1883)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **bypass_author_approval**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/602fc39ddabeb6f87f67986103490d69cf5205ee) did not succeed in branch w/7.5/feature/S3C-1139-sproxyd-batchdelete.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/cloudserver/commit/fee90d8052faa734c7a629a02088c5349d4dea9b) did not succeed in branch w/8.0/feature/S3C-1139-sproxyd-batchdelete.




*The following options are set:* **bypass_author_approval**

rahulreddy commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **bypass_author_approval**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.5/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.4/feature/S3C-1139-sproxyd-batchdelete`
and `development/7.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.5/feature/S3C-1139-sproxyd-batchdelete origin/development/7.5
 $ git merge origin/w/7.4/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.5/feature/S3C-1139-sproxyd-batchdelete
```





*The following options are set:* **bypass_author_approval**

rahulreddy committed:
improvement: implement batch delete for sproxydclient

This commit implements the sproxyd's batch delete method in the
data abstractions so that APIs can leverage it for deleting large
number of keys. It also changes the batchDelete function's signature
by adding a callback. It is expected for the api methods to use callback
when invoking the method.

bert-e commented:
# History mismatch



Merge commit #8f49fd8d14cf0c41bb0a24da1e3bcb1199e4fe4a on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

rahulreddy commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

rahulreddy commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/S3C-1139-sproxyd-batchdelete origin/development/8.1
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# History mismatch



Merge commit #481b2b62811f38fe87165e7cd856ec1eca715af6 on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bennettbuchanan commented on lib/services.js:
```diff
@@ -250,16 +250,21 @@ const services = {
                     if (err) {
                         return cb(err, res);
                     }
-                    cb(null, res); // this is smart
```
Was the idea here to allow the batch delete to happen in the background? Is it possible, then, with large batch delete operations the request will timeout?

rahulreddy commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

bennettbuchanan commented on lib/data/multipleBackendGateway.js:
```diff
@@ -117,6 +117,14 @@ const multipleBackendGateway = {
+        if (client.batchDelete) {
+            return client.batchDelete(keys, log.getSerializedUids(), callback);
+        }
+        return callback(errors.NotImplemented);
```
Same comment as @philipyoo: https://github.com/scality/Arsenal/pull/808/files#r289941778.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/S3C-1139-sproxyd-batchdelete origin/development/8.1
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/S3C-1139-sproxyd-batchdelete` into
`stabilization/7.4.4`
* [w/7.4/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.4...w/7.4/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.4` (pull request #1884)
* [w/7.5/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.5...w/7.5/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.5` (pull request #1885)
* [w/8.0/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.0...w/8.0/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.0` (pull request #1886)
* [w/8.1/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.1` (pull request #1887)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @rachedbenmustapha

* @rahulreddy

* @dora-korpar

bert-e commented:
# History mismatch



Merge commit #39bb67d16e5e310e4c50e6ec05e8dd510c91b393 on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

rahulreddy commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**

bert-e commented:
# History mismatch



Merge commit #39bb67d16e5e310e4c50e6ec05e8dd510c91b393 on the integration branch
`w/7.4/feature/S3C-1139-sproxyd-batchdelete` is merging a branch which is neither the current
branch `feature/S3C-1139-sproxyd-batchdelete` nor the development branch
`development/7.4`.

It is likely due to a rebase of the branch `feature/S3C-1139-sproxyd-batchdelete` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

rahulreddy commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.0/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/7.5/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.0`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.0/feature/S3C-1139-sproxyd-batchdelete origin/development/8.0
 $ git merge origin/w/7.5/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.0/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/feature/S3C-1139-sproxyd-batchdelete` with contents from `w/8.0/feature/S3C-1139-sproxyd-batchdelete`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/feature/S3C-1139-sproxyd-batchdelete origin/development/8.1
 $ git merge origin/w/8.0/feature/S3C-1139-sproxyd-batchdelete
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/feature/S3C-1139-sproxyd-batchdelete
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/S3C-1139-sproxyd-batchdelete` into
`stabilization/7.4.4`
* [w/7.4/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.4...w/7.4/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.4` (pull request #1889)
* [w/7.5/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/7.5...w/7.5/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/7.5` (pull request #1890)
* [w/8.0/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.0...w/8.0/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.0` (pull request #1891)
* [w/8.1/feature/S3C-1139-sproxyd-batchdelete](https://github.com/scality/cloudserver/compare/development/8.1...w/8.1/feature/S3C-1139-sproxyd-batchdelete) will be merged into `development/8.1` (pull request #1892)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @rachedbenmustapha

* @rahulreddy

* @dora-korpar

rahulreddy commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `stabilization/7.4.4`

* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.5`

* :heavy_check_mark: `development/8.0`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `stabilization/7.4.4`

* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.5`

* :heavy_check_mark: `development/8.0`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`



Please check the status of the associated issue S3C-1139.

Goodbye rahulreddy.
